torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -448,6 +448,28 @@ def rewrite_node(node):
             args = node.args
             kwargs = node.kwargs
 
+            def batch_norm_post_process(output, shape, weight, bias):
+                if weight is not None:
+                    weight = g.call_function(ttnn.reshape, (weight, shape))
+                    output = g.call_function(ttnn.mul, (output, weight))
+                if bias is not None:
+                    bias = g.call_function(ttnn.reshape, (bias, shape))
+                    output = g.call_function(ttnn.add, (output, bias))
+                return output
+
+            # Non-training batch normalization
+            def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
+                assert is_getitem_0_only_user(node), "non-training batch_norm should only has first return value used"
+                shape = input.meta["val"].size()
+                shape = (1, shape[1]) + (1,) * (len(shape) - 2)
+                invstd = g.call_function(ttnn.rsqrt, (g.call_function(ttnn.add, (var, eps)),))
+                invstd = g.call_function(ttnn.reshape, (invstd, shape))
+                mean = g.call_function(ttnn.reshape, (mean, shape))
+                output = g.call_function(ttnn.sub, (input, mean))
+                output = g.call_function(ttnn.mul, (output, invstd))
+                output = batch_norm_post_process(output, shape, weight, bias)
+                return g.call_function(target_wrappers.pack_to_tuple, (output,))
```

Comment:
Want to highlight that this code could be contributed to TT-NN as a C++ composite operation.

Reply:
@jdh8 Could you help open an issue in tt-metal for this improvement?

Reply:
tenstorrent/tt-metal#15449